server/src/main/java/org/opensearch/common/cache/store/OpenSearchOnHeapCache.java

```diff
@@ -121,7 +129,11 @@ public Iterable<ICacheKey<K>> keys() {
 
     @Override
     public long count() {
-        return cacheStatsHolder.count();
+        if (useNoopStats) {
+            return cache.count();
+        } else {
+            return cacheStatsHolder.count();
+        }
```

Comment:
Could this use `cache.count()` always? If the `Cache` has the count, then we don't need to go to the `CacheStatsHolder` to get it.